sequence_processing_pipeline/Pipeline.py

```diff
@@ -360,20 +361,37 @@ def get_sample_ids(self):
 
         return results
 
-    def get_sample_names(self):
+    def get_sample_names(self, project_name=None):
         '''
         Returns list of sample-names sourced from sample-sheet or pre-prep file
+        :param project_name: If None, return all sample-names.
         :return: list of sample-names
         '''
-
         # test for self.mapping_file, since self.sample_sheet will be
         # defined in both cases.
         if self.mapping_file is not None:
-            results = list(self.mapping_file.sample_name.unique())
+            return self._get_sample_names_from_mapping_file(project_name)
         else:
-            results = [x.Sample_Name for x in self.sample_sheet.samples]
+            return self._get_sample_names_from_sample_sheet(project_name)
 
-        return results
+    def _get_sample_names_from_sample_sheet(self, project_name):
+        if project_name is None:
+            return [x.Sample_Name for x in self.sample_sheet.samples]
+        else:
+            # Since the project-name is stored in an internal variable
+            # in a third-party library, convert the data structure to
+            # JSON using the exposed method and obtain from the result.
+            jsn = json_loads(self.sample_sheet.to_json())
+            return [x['Sample_Name'] for x in jsn['Data'] if
+                    f'{project_name}_' in x['Sample_Project']]
```

Comment:
I'm a little lost about this, can you point me to the test for this as I think that should shed some light to understand what's going on here?

Reply:
The test is an addition to test_get_sample_names() in test_Pipeline.py; line 1382 to be specific. In addition to testing that all of the sample-names in the sample-sheet are returned, it now tests to see that only the nine entries of the Gerwick project are returned when that project is specified.

Reply:
There's a minor complication in that the metadata in the sample-sheet uses the fully qualified project-name with Qiita ID while it's more user-friendly to use just the project name. Rather than require the fully qualified project-name with Qiita ID, f'{project_name}_' in x['Sample_Project'] is used to look for just the project name within the value of that column. I don't prefer this method but I think it's very unlikely that project-name_ is going to be a subset of another project-name.